src/commands/code/secret.ts

```diff
@@ -0,0 +1,127 @@
+  if (!mode) {
+    mode = await PromptService.promptSelectionWithAutoComplete<APP_SECRET_MANAGEMENT_MODES>(
+      'Select app secret variables management mode',
+      Object.values(APP_SECRET_MANAGEMENT_MODES),
+    );
+  }
+
+  return mode;
+};
+
+const promptForKeyIfNotProvided = async (mode: APP_SECRET_MANAGEMENT_MODES, appId: AppId, key?: string) => {
+  if (!key && isKeyRequired(mode)) {
+    const existingKeys = await listAppSecretKeys(appId);
+    key = await PromptService.promptSelectionWithAutoComplete('Enter key for secret variable', existingKeys, {
+      includeInputInSelection: true,
+    });
+  }
+
+  return key;
+};
+
+const promptForValueIfNotProvided = async (mode: APP_SECRET_MANAGEMENT_MODES, value?: string) => {
+  if (!value && isValueRequired(mode)) {
+    value = await PromptService.promptForHiddenInput(
+      'value',
+      'Enter value for secret variable',
+      'You must enter a value value',
+    );
+  }
+
+  return value;
+};
+
+const flagsWithModeRelationships: Relationship = {
+  type: 'all',
+  flags: [
+    {
+      name: 'mode',
+
+      when: async (flags: Record<string, unknown>) => isValueRequired(flags.mode as (typeof MODES_WITH_KEYS)[number]),
+    },
+  ],
+};
+
+export default class Secret extends AuthenticatedCommand {
+  static description = 'Manage secret variables for your app hosted on monday-code.';
+
+  static examples = ['<%= config.bin %> <%= command.id %>'];
+
+  static flags = Secret.serializeFlags(
+    addRegionToFlags({
+      appId: Flags.integer({
+        char: 'i',
+        aliases: ['a'],
+        description: 'The id of the app to manage secret variables for',
+      }),
+      mode: Flags.string({
+        char: 'm',
+        description: 'management mode',
+        options: Object.values(APP_SECRET_MANAGEMENT_MODES),
+      }),
+      key: Flags.string({
+        char: 'k',
+        description: 'variable key [required for set and delete]]',
+        relationships: [flagsWithModeRelationships],
+      }),
+      value: Flags.string({
+        char: 'v',
+        description: 'variable value [required for set]',
+        relationships: [flagsWithModeRelationships],
+      }),
+    }),
+  );
+
+  static args = {};
+  DEBUG_TAG = 'secret';
+  public async run(): Promise<void> {
+    try {
+      const { flags } = await this.parse(Secret);
+      const { region: strRegion } = flags;
+      const region = getRegionFromString(strRegion);
+      let { mode, key, value, appId } = flags as ManageAppSecretFlags;
+
+      if (!appId) {
+        appId = Number(await DynamicChoicesService.chooseApp());
+      }
+
+      const selectedRegion = await chooseRegionIfNeeded(region, { appId });
+
+      mode = await promptForModeIfNotProvided(mode);
+      key = await promptForKeyIfNotProvided(mode, appId, key);
+      value = await promptForValueIfNotProvided(mode, value);
+      this.preparePrintCommand(this, { appId, mode, key, value, region: selectedRegion });
+      console.log("1")
```

Comment:
as you can see here, and further down the road, it's a best practice to use the commit modal in Webstorm and "cherry pick" specific changes that would go on the commit, reverting unwanted things like console logs or leaving them un-commited locally just for debugging.
If you're not familiar with that, lets sit together for 10 mins.

also, you see our code base has automatically enforced standarts- below you see an automatic message for lint/prettier errors - using double quotes "" instead of single quotes ''.

But anyway, to "pass the code-review" you'll have to do some fixes, start by removing all the unneeded console logs.

Reply:
Got it, i'm writing to me to talk with you about the "cherry pick".